File: 1063-best-sightseeing-pair/1063-best-sightseeing-pair.js
/**
 * @param {number[]} values
 * @return {number}
 */
var maxScoreSightseeingPair = function(values) {
    let ans = -Infinity, curr = values[0]
    for(let i = 1; i<values.length; i++) {
        ans = Math.max(ans, curr + values[i] - i);
        curr = Math.max(curr, values[i] + i);
    }

    return ans;
};

File: 1063-best-sightseeing-pair/1063-best-sightseeing-pair.java
class Solution {
    public int maxScoreSightseeingPair(int[] values) {
        int ans = Integer.MIN_VALUE, curr = values[0];
        for(int i = 1; i<values.length; i++) {
            ans = Math.max(ans, curr + values[i] - i);
            curr = Math.max(curr, values[i] + i);
        }

        return ans;
    }
}